### PR TITLE
Clone to closure idiom

### PR DESCRIPTION
**Pull request by kriomant**

### CODE REVIEW

File: idioms/clone-to-closure.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+to move environment. However, often you want to give copy of some data to closure.
+Use variable rebinding in separate scope for that.
+
+
+## Example
+
+```rust
+let mut num = Rc::new(5);
+let closure = {
+	let num = num.clone();
```
Comment (lfairy) [inline idioms/clone-to-closure.md]: This code should use four-space indents, not tabs, to stay consistent with other articles.

File: idioms/clone-to-closure.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+
+## Advantages
+
+Copied data are grouped together with closure definition, so their purpose is more clear
+and they will be dropped immediately even if they are not consumed by closure.
+
+Closure uses same variable names as surrounding code whether data are copied or moved.
+
```
Comment (lfairy) [inline idioms/clone-to-closure.md]: As a disadvantage, you should mention rightward drift: adding another scope will indent the code further, which can make it harder to read.

File: idioms/clone-to-closure.md (hunk 1 above)
Comment (kriomant) [inline idioms/clone-to-closure.md]: fixed

File: idioms/clone-to-closure.md (hunk 2 above)
Comment (kriomant) [inline idioms/clone-to-closure.md]: done

**Comment by nrc** [comment]
I think the name of the pattern is a little misleading - it applies just as much to by reference variables captured by a `move` closure as it does to cloned ones.

File: idioms/pass-var-to-closure.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
-# Pass clone to closure
+# Pass variable to closure
```
Comment (lfairy) [inline idioms/pass-var-to-closure.md]: This should be "variables"

**Comment by lfairy** [comment]
Looks good to me. Since we've had two people review it already, I'll go ahead and merge.

Thanks @kriomant for your contribution!

**Pull request merged**